include/sway/layers.h

```diff
@@ -24,6 +24,8 @@ struct sway_layer_surface {
 
 	struct wlr_box geo;
 	enum zwlr_layer_shell_v1_layer layer;
+
+	struct wl_list input_popups;
```

Comment:
Hm, why is there a list of input popups for each layer surface?

I figured input popups would be "global", not per-layer-surface. Another example of a "global" surface is drag icons.

Reply:
I'm not sure if this review comment is specific to layer surfaces or includes views too, but if it's both, then:
It seems to me that having the input popup in the view/layer surface is simpler than having it global, because it needs to be positioned relative to the parent surface. If it isn't, then the popup has to be updated every time the parent surface changes.

Reply:
Hm, but the IME only creates a single popup, right? And it's re-used for all text-input clients?

When the parent surface changes, the popup needs to be updated anyways, because the focused text field changes as well?

Reply:
> Hm, but the IME only creates a single popup, right? And it's re-used for all text-input clients?

With the current protocol, input methods can create multiple popups, but without positioning support, this is a bit useless.
But I'm not sure if there's a usecase for multiple popups (on the same seat) even with positioning support.

> When the parent surface changes, the popup needs to be updated anyways, because the focused text field changes as well?

My wording wasn't clear, I meant the parent surface updating geometry, not the focus changing.

Reply:
> But I'm not sure if there's a usecase for multiple popups (on the same seat) even with positioning support.

Wondered about this as well: https://gitlab.freedesktop.org/wayland/wayland-protocols/-/issues/40#note_847039

If the Sway implementation can be simplified by ignoring extra popups, maybe it'd be worth it.

> I meant the parent surface updating geometry, not the focus changing.

We can get the globally-active IME. From the IME we can get the `wlr_surface` with the focused text field. If we could get the position of this surface, then positioning the IME popup would be easy, correct?